sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -162,9 +162,11 @@ case class ShuffleQueryStageExec(
     }
   }
 
-  def mapStats: MapOutputStatistics = {
-    assert(resultOption.isDefined)
-    resultOption.get.asInstanceOf[MapOutputStatistics]
+  def mapStats: Option[MapOutputStatistics] = {
+    assert(resultOption.isDefined, "ShuffleQueryStageExec should already be ready")
+    val stats = resultOption.get.asInstanceOf[MapOutputStatistics]
+    // mapStats can be null when the input RDD of a plan has 0 partition
```

Comment:
nit: `mapStats` -> `stats`

Reply:
can we make the comment as a method doc?

Reply:
e.g.
```
/**
 * Returns the .... If the shuffle map stage has no partition, this method returns None, as there is no map statistics.
 */
def mapStats: Option[MapOutputStatistics] = {
  assert(resultOption.isDefined, "ShuffleQueryStageExec should already be ready")
  Option(resultOption.get.asInstanceOf[MapOutputStatistics])
}
```